### DISCUSSION TITLE
Better debug mode for biome to understand why it's slow

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#518] [Ideas]
I see for some files it takes some time for biome to run.

<img width="602" alt="Screenshot 2023-10-13 at 10 05 35" src="https://github.com/biomejs/biome/assets/9585787/c10526a6-dec9-4022-89ae-f4d3485f009f">

time was about 20 seconds, disabling autofix brought it down to 10 but 10 already seems a bit long.

Would be good to get an output from biome showing:

- which lines of code take the most processing time
- how processing time is divided (parsing? auto-fixing?)
- which rules are taking up most time?

**Comment by ematipico** [comment]
Can you expand about **how** you would expect to gather this information?

**Reply by jpike88** [reply]
Create a statistics hashmap for the run, and fill it up appropriately. Then dump that object to debug console. Similar to how any kind of debug statistics collection is handled I guess.

**Reply by ematipico** [reply]
Sorry, probably I wasn't clear enough.

How would you want to collect this information? Using the CLI or the VSCode extension? If using the CLI, how? A command , some new options, etc. If using the extension, how? 

And how should this information appear? 

The proposal is very broad, and I would like to understand how a user expects to use this "debug mode".

**Reply by jpike88** [reply]
Ah right, for something like that I'd expect to see debug info through the Output panel in VSCode
